### PR TITLE
[SVS] add singing tacotron

### PR DESCRIPTION
**Pull request by A-Quarter-Mile**
Hi @ftshijt, here are updates on Muskits.

- Add Singing Tacotron in svs with GDCA & dynamic filter.
- Fix score_scp.
- Fix CI.
- Correct feature naming: tempo -> duration.

### CODE REVIEW

**Comment by ftshijt** [comment]
Is the performance stable for now?

File: egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+sort_batch: descending      # how to sort created batches
+num_workers: 10             # number of workers of data loader
+train_dtype: float32        # dtype in training
+log_interval: 1             # log interval in iterations
+keep_nbest_models: 2        # number of models to keep
+num_att_plot: 3             # number of attention figures to be saved in every check
+seed: 0                     # random seed number
+best_model_criterion:
+-   - valid
+    - loss
```
Comment (ftshijt) [inline egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml:82]: Have you tried instead use total_counts as the criterion?

File: espnet2/fileio/score_scp.py [hunk 2]
```diff
@@ -83,12 +83,20 @@ def __getitem__(self, key):
...
+                    st += dur
+                    continue
                 else:  # normal note for one syllable
                     notes_list.append(NOTE(lr, note.pitch.midi, st, st + dur))
                 prepitch = note.pitch.midi
-                for arti in note.articulations:  # <br> is tagged as a notation
-                    if arti.name in ["breath mark"]:  # up-bow?
-                        notes_list.append(NOTE("B", 0, st, st))  # , 0))
+                for arti in note.articulations:
+                    # NOTE(Yuning): By default, 'breath mark' appears at the end of
```
Comment (ftshijt) [inline espnet2/fileio/score_scp.py:92]: I think this is worth noting. Can you also add this info to the doc as well

File: espnet2/svs/singing_tacotron/decoder.py [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+from espnet.nets.pytorch_backend.tacotron2.decoder import Postnet, Prenet, ZoneOutCell
+
+
+def decoder_init(m):
+    """Initialize decoder parameters."""
+    if isinstance(m, torch.nn.Conv1d):
+        torch.nn.init.xavier_uniform_(m.weight, torch.nn.init.calculate_gain("tanh"))
+
+
+class Decoder(torch.nn.Module):
```
Comment (ftshijt) [inline espnet2/svs/singing_tacotron/decoder.py:24]: Since this is very similar to tacotron, do you think if we are possible to inherit the tactotron decoder to minimize the code usage?

File: espnet2/svs/singing_tacotron/encoder.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence
+
+
+def encoder_init(m):
+    """Initialize encoder parameters."""
+    if isinstance(m, torch.nn.Conv1d):
+        torch.nn.init.xavier_uniform_(m.weight, torch.nn.init.calculate_gain("relu"))
+
+
+class Encoder(torch.nn.Module):
```
Comment (ftshijt) [inline espnet2/svs/singing_tacotron/encoder.py:20]: Ditto, if possible, let's inherit Tomoki's original one

**Review by ftshijt — COMMENTED** [review]
LGTM! Some minor suggestions.

File: espnet2/svs/singing_tacotron/encoder.py (hunk 4 above)
Comment (A-Quarter-Mile) [reply espnet2/svs/singing_tacotron/encoder.py:20]: For decoder, the transition token in GDCA makes it different from tacotron's.

File: espnet2/svs/singing_tacotron/decoder.py (hunk 3 above)
Comment (A-Quarter-Mile) [reply espnet2/svs/singing_tacotron/decoder.py:24]: For decoder, the transition token in GDCA makes it different from tacotron's.

File: espnet2/svs/singing_tacotron/encoder.py (hunk 4 above)
Comment (A-Quarter-Mile) [reply espnet2/svs/singing_tacotron/encoder.py:20]: Tacotron's encoder gets text as input and we use content embedding as input. The content embedding can be different for GDCA and others. Besides, a new duration_encoder is added here.

File: egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml (hunk 1 above)
Comment (A-Quarter-Mile) [reply egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml:82]: I haven't tried this.

File: espnet2/fileio/score_scp.py (hunk 2 above)
Comment (A-Quarter-Mile) [reply espnet2/fileio/score_scp.py:92]: Fixed.

File: egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml (hunk 1 above)
Comment (ftshijt) [reply egs2/opencpop/svs1/conf/tuning/train_singing_tacotron.yaml:82]: Yeah, for TTS model, it might be good to use total_count (maybe we can have a check if that would be better) as loss is not a very good metric

**Comment by ftshijt** [comment]
You might have known. FYI: you may check https://app.codecov.io/gh/espnet/espnet/tree/master/espnet2%2Fsvs for the coverage and where to add for CI test

Our target is 80% if possible

**Comment by ftshijt** [comment]
LGTM! Thanks for your contribution.
